dask/array/learn.py

```diff
@@ -75,10 +76,10 @@ def fit(model, x, y, get=threaded.get, **kwargs):
     dsk = {(name, -1): model}
     dsk.update(dict(((name, i), (_partial_fit, (name, i - 1),
                                                (x.name, i, 0),
-                                               (y.name, i), kwargs))
+                                               (getattr(y, 'name', ''), i), kwargs))
```

Comment:
Does partial_fit always allow an empty argument or does it sometimes expect nothing?

Reply:
On the scikit-learn side, all estimators have to take an `X` and a `y`. For unsupervised ones like `MiniBatchKMeans`, the `y` is ignored.